src/main/java/gregtech/api/worldgen/bedrockFluids/BedrockFluidVeinHandler.java

```diff
@@ -135,39 +148,30 @@ public static int getDepletedFluidRate(World world, int chunkX, int chunkZ) {
     /**
      * Depletes fluid from a given chunk
      *
-     * @param world  World whose chunk to drain
-     * @param chunkX Chunk x
-     * @param chunkZ Chunk z
+     * @param coefficient the depletion coefficient of the rig depleting the chunk
+     * @param world       World whose chunk to drain
+     * @param chunkX      Chunk x
+     * @param chunkZ      Chunk z
      */
-    public static void depleteVein(World world, int chunkX, int chunkZ) {
-        if (world.isRemote)
-            return;
-
+    public static void depleteVein(float coefficient, World world, int chunkX, int chunkZ) {
         FluidVeinWorldEntry info = getFluidVeinWorldEntry(world, chunkX, chunkZ);
         if (info == null || info.currentFluidAmount == 0)
             return;
-
         BedrockFluidDepositDefinition definition = info.getVein();
 
-        // attempt to deplete the vein
-        // ciel(chance) - chance * 100 = % to get depleted
-        // if random value < % chance, deplete by 1
-//        int rawChance = definition.getDepletionChance();
-//        int chanceToDecrease = (int) (100 * (Math.ceil(rawChance) - rawChance));
-//        if (GTUtility.getRandomIntXSTR(100) < chanceToDecrease)
-
-        // alternative depletion algorithm: 1 in vein's chance to deplete by vein's depletion amount
-        if (GTValues.RNG.nextInt(definition.getDepletionChance()) == 1)
+        double averageDecrease = definition.getDepletionChance() * definition.getDepletionAmount() * coefficient;
+        int decrease = (int) Math.ceil(averageDecrease);
+        if (GTValues.RNG.nextFloat() < (decrease - averageDecrease)) {
```

Comment:
This seems like a complicated and non functional way of preventing over-draining the vein.
I propose this:
```java
int decrease = (int) Math.floor(averageDecrease + GTValues.RNG.nextFloat());
if (decrease > 0) {
    info.currentFluidAmount = Math.max(0, info.currentFluidAmount - decrease);
    BedrockFluidVeinSaveData.setDirty();
}
```